rest_api/index.adoc

```diff
@@ -7,8 +7,211 @@
+Namespace:	test
+Labels:		<none>
+Annotations:	kubernetes.io/service-account.name=robot,kubernetes.io/service-account.uid=ea70e4c7-0663-11e6-b279-fa163e610e01
+
+Type:	kubernetes.io/service-account-token
+
+Data
+====
+token:		fyJhbGciOiJSUzI1NiIyInR5cCI2IkpXVCJ9...
+ca.crt:		1070 bytes
+namespace:	8 bytes
+----
+
+The token value may be used as an in an authorization header to
+link:#examples[authenticate API calls], the
+link:../dev_guide/service_accounts.html#using-a-service-account-s-credentials-externally[CLI]
+or in the link:#docker-login[docker login command]. Service accounts may be
+created and deleted as needed with the appropriate role(s) assigned. See the
+link:../architecture/additional_concepts/authorization.html#roles[authorization section]
+for a discussion on roles.
+
+== Examples
+
+These examples are provided as a reference to provide quick success making rest
+API calls. They use insecure methods. In these examples a simple **GET** call
+is made to link:openshift_v1.html#get-available-resources[list
+available resources].
+
+=== cURL
+
+.Request (insecure)
+====
+----
+$ curl -X GET -H "Authorization: Bearer <token>" https://openshift.redhat.com:8443/oapi/v1 --insecure
+----
+====
+
+.Result (truncated)
+====
+----
+{
+  "kind": "APIResourceList",
+  "groupVersion": "v1",
+  "resources": [
+    {
+      "name": "buildconfigs",
+      "namespaced": true,
+      "kind": "BuildConfig"
+    },
+    {
+      "name": "buildconfigs/instantiate",
+      "namespaced": true,
+      "kind": "BuildRequest"
+    },
+    {
+      "name": "buildconfigs/instantiatebinary",
+      "namespaced": true,
+      "kind": "BinaryBuildRequestOptions"
+    },
+    {
+      "name": "buildconfigs/webhooks",
+      "namespaced": true,
+      "kind": "Status"
+    },
+    {
+      "name": "builds",
+      "namespaced": true,
+      "kind": "Build"
+    },
+    ...
+    {
+      "name": "subjectaccessreviews",
+      "namespaced": true,
+      "kind": "SubjectAccessReview"
+    },
+    {
+      "name": "templates",
+      "namespaced": true,
+      "kind": "Template"
+    },
+    {
+      "name": "useridentitymappings",
+      "namespaced": false,
+      "kind": "UserIdentityMapping"
+    },
+    {
+      "name": "users",
+      "namespaced": false,
+      "kind": "User"
+    }
+  ]
+}
+----
+====
+
+
+=== Python
+
+.Interactive Python API call using "requests" module (insecure)
+====
+----
+>>> import requests
+>>> url = 'https://openshift.redhat.com:8443/oapi/v1'
+>>> headers = {'Authorization': 'Bearer dIAo76N-W-GXK3S_w_KsC6DmH3MzP79zq7jbMQvCOUo'}
+>>> requests.get(url, headers=headers, verify=False)
+/usr/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
+  InsecureRequestWarning)
+<Response [200]>
+----
+====
+
+=== Docker Login
```

Comment:
@adellape I added this docker login section. Not sure it goes here but wanted to get it in. Is there a better place or does this work for now? I'm particularly interested in covering this topic for openshift online to cover the use case where users build docker images and push them to openshift online registry.